osu.Game.Tests/Chat/TestSceneChannelManager.cs

```diff
@@ -83,12 +89,62 @@ public void TestCommandsPostedToCorrectChannelWhenNotCurrent()
             AddAssert("/np command received by channel 2", () => channel2.Messages.Last().Content.Contains("is listening to"));
         }
 
+        [Test]
+        public void TestMarkAsReadIgnoringLocalMessages()
+        {
+            Channel channel = null;
+
+            AddStep("join channel and select it", () =>
+            {
+                channelManager.JoinChannel(channel = createChannel(1, ChannelType.Public));
+                channelManager.CurrentChannel.Value = channel;
+            });
+
+            AddStep("post message", () => channelManager.PostMessage("Something interesting"));
+            AddUntilStep("wait until the message is posted", () => channel.Messages.Count == sentMessages.Count);
+            AddStep("post /help command", () => channelManager.PostCommand("help", channel));
+            AddStep("post /me command with no action", () => channelManager.PostCommand("me", channel));
+            AddStep("post /join command with no channel", () => channelManager.PostCommand("join", channel));
+            AddStep("post /join command with non-existent channel", () => channelManager.PostCommand("join i-dont-exist", channel));
+            AddStep("post non-existent command", () => channelManager.PostCommand("non-existent-cmd arg", channel));
+
+            AddStep("register mark channel as read request handler", () =>
+            {
+                ((DummyAPIAccess)API).HandleRequest = req =>
+                {
+                    switch (req)
+                    {
+                        case MarkChannelAsReadRequest markRead:
```

Comment:
Could this not be together with the rest of the handler? Preferably after splitting out method or methods as it'd be getting a little large, but the other handler doesn't handle this message type, so it should be fine to merge...?

Reply:
It was down here only because of the assert. I moved the assert to the top scope and merged the request handlers.